lib/vanagon/driver.rb

```diff
@@ -13,17 +13,21 @@ class Driver
     attr_accessor :platform, :project, :target, :workdir, :verbose, :preserve
     attr_accessor :timeout, :retry_count
 
-    def initialize(platform, project, options = { :configdir => nil, :target => nil, :engine => nil, :components => nil, :skipcheck => false, :verbose => false, :preserve => false }) # rubocop:disable Metrics/AbcSize
+    def initialize(platform, project, options = { :configdir => nil, :target => nil, :engine => nil, :components => nil, :skipcheck => false, :verbose => false, :preserve => false, :only_build => nil }) # rubocop:disable Metrics/AbcSize
       @verbose = options[:verbose]
       @preserve = options[:preserve]
 
       @@configdir = options[:configdir] || File.join(Dir.pwd, "configs")
       components = options[:components] || []
+      only_build = options[:only_build] || nil
       target = options[:target]
       engine = options[:engine] || 'pooler'
 
       @platform = Vanagon::Platform.load_platform(platform, File.join(@@configdir, "platforms"))
       @project = Vanagon::Project.load_project(project, File.join(@@configdir, "projects"), @platform, components)
+      if only_build
+        filter_out_components(only_build)
+      end
```

Comment:
It's a nitpick, but lines 28-30 could probably be moved under the current line 32 (`@project.settings[:skipcheck] = options[:skipcheck]`), so that filtering happens after all of the settings are initialized (filtering shouldn't depend on settings one way or the other). You could also collapse the `if` into a single line for the sake of conciseness:

```ruby
filter_out_components(only_build) if only_build
```

(I like using this approach when possible because it makes poor variable name choices immediately obvious when your brain tries to parse the instructions as plain English and balks at bad naming choices.)